### PR TITLE
invidious: unstable-2022-03-16 -> unstable-2022-05-04

### PR DESCRIPTION
**Pull request by schnusch**
###### Description of changes

Upstream update.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [x] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by infinisil** [comment]
CI fails to build invidious (or rather, lsquic)

**Comment by schnusch** [comment]
There haven't been any changes to `invidious.lsquic` or `invidious.lsquic.boringssl` in a while and I successfully built it on x86_64-linux and aarch64-linux.

Also `nix-build --check -A invidious.lsquic.boringssl` and `nix-build --check -A invidious.lsquic` are successful.

**Comment by infinisil** [comment]
While I can't reproduce the CI failure from this PR's commit directly, I can reproduce it after a merge with master, which I think ofborg does automatically. No idea why it fails but it you should be able to bisect it: c418d051908a692ec69960fc09c2cb70e6c570e3 (current PR base) works, but e9d8ace0ae0466f59419bca4dffa4dd433e899c3 (current master) breaks

File: pkgs/servers/invidious/lsquic.nix [hunk 1]
```diff
@@ -13,7 +23,40 @@ let
     patches = [
       # Use /etc/ssl/certs/ca-certificates.crt instead of /etc/ssl/cert.pem
       ./use-etc-ssl-certs.patch
+
+      # fix gcc11 build errors
+      (fetchGitilesPatch {
+        name = "fix-mismatch-between-header-and-implementation-of-bn_sqr_comba8.patch";
+        url = "https://boringssl.googlesource.com/boringssl/+/139adff9b27eaf0bdaac664ec4c9a7db2fe3f920";
```
Comment (infinisil) [inline pkgs/servers/invidious/lsquic.nix:30]: Are all these patches in boringssl master? If so, why not just update boringssl?

**Comment by schnusch** [comment]
lsquic requires a specific version of boringssl, see <https://github.com/litespeedtech/lsquic#Requirements>.

I am not sure if this is really necessary and no newer boringssl version can be used, but I cannot judge that.

**Comment by infinisil** [comment]
I see, so it's that lsquic wants a specific version, but some patches from boringssl master are required to fix it for GCC 11. I think the comments for these patches could be improved a bit to explain this.

**Comment by infinisil** [comment]
Looks like lsquic got fixed with a downgrade to GCC 10 by @squalus in https://github.com/NixOS/nixpkgs/pull/172782, which is now causing merge conflicts. @schnusch What do you think is the better solution here?

**Comment by schnusch** [comment]
I will rebase it onto #172782's HEAD.

Using GCC10 seems easier, but 3/4 of errors are fixed quite easily and perhaps will not be necessary when lsquic/boringssl will be updated. I think it's nicer to avoid multiple GCC versions.

**Review by infinisil — APPROVED** [review]
Built and tested, thanks!

**Pull request merged**